handler/src/main/java/io/netty/handler/pcap/PcapWriteHandler.java

```diff
@@ -153,8 +163,44 @@ public PcapWriteHandler(OutputStream outputStream, boolean captureZeroByte, bool
         this.writePcapGlobalHeader = writePcapGlobalHeader;
     }
 
-    @Override
-    public void channelActive(ChannelHandlerContext ctx) throws Exception {
+    /**
+     * Force this handler to write data as if they were TCP packets, with the given connection metadata. If this method
+     * isn't called, we determine the metadata from the channel.
+     *
+     * @param serverAddress The address of the TCP server (handler)
+     * @param clientAddress The address of the TCP client (initiator)
+     * @param localIsServer Whether the handler is part of the server channel
+     */
+    public void forceTcpChannel(
+            InetSocketAddress serverAddress,
+            InetSocketAddress clientAddress,
+            boolean localIsServer) {
+        channelType = ChannelType.TCP;
+        handlerAddr = serverAddress;
+        initiatiorAddr = clientAddress;
```

Comment:
Generally speaking all the variables that are changed within here need to be volatile as these are accessed from the  EventLoop and also from another thread (at least it could). To ensure we correctly handle this without races I would suggest to group these in an extra internal class and store the instance as a volatile. This way we ensure we always update all together.

Reply:
@normanmaurer my intention here was that the user would call these methods once before adding the handler to the pipeline. i dont think it makes sense to call this method concurrently.

i'm going to change the code to protect against this

Reply:
if this is the case then let the user do these things via the constructor

Reply:
makes sense, i will make a builder